hibernate-core/src/main/java/org/hibernate/engine/query/spi/QueryPlanCache.java

```diff
@@ -148,13 +149,28 @@ public HQLQueryPlan getHQLQueryPlan(String queryString, boolean shallow, Map<Str
 			throws QueryException, MappingException {
 		final HQLQueryPlanKey key = new HQLQueryPlanKey( queryString, shallow, enabledFilters );
 		HQLQueryPlan value = (HQLQueryPlan) queryPlanCache.get( key );
+		boolean stats = factory.getStatistics().isStatisticsEnabled();
+
 		if ( value == null ) {
+			final long startTime = System.nanoTime();
```

Comment:
It would be nice if we could skip that when statistics are not enabled.

`System.nanoTime()` adds some overhead.